pipeline_steps/common.groovy

```diff
@@ -492,24 +492,24 @@ def is_doc_update_pr(git_dir) {
  * This function uses environment variables injected by github pull request
  * builder and so can only be used for PR triggered jobs
  */
-def get_jira_issue_key(repo_path="rpc-openstack"){
-  def key_regex = "^[a-zA-Z][a-zA-Z0-9_]+-[1-9][0-9]*"
+def get_jira_issue_key(repo_org="rcbops", repo_path="rpc-openstack"){
+  def key_regex = "[a-zA-Z][a-zA-Z0-9_]+-[1-9][0-9]*"
   dir(repo_path){
-    commit_titles = sh(
+    commits = sh(
       returnStdout: true,
-      script: "git log --pretty=%s origin/${ghprbTargetBranch}..${ghprbSourceBranch}").split('\n')
-    print("looking for Jira issue keys in the following commits: ${commit_titles}")
-    for (def i=0; i<=commit_titles.size(); i++){
-      try{
-        key = (commit_titles[i] =~ key_regex)[0]
-        print ("Found Jira Issue Key: ${key}")
-        return key
-      } catch (e){
-        continue
-      }
+      script: """
+      git remote add upstream https://github.com/${repo_org}/${repo_path}.git 2>&1 > /dev/null
```

Comment:
I would put this and the next line in github.add_issue_url_to_pr after the `git` call, then the repo clone, branch checkout and upstream addition are done in the same place. Add_issue_url_to_pr handles the repo setup, and common.get_jira_issue_key assumes the repo is correctly setup and diffs the two refs. It could take an upstream parameter that defaults to `upstream`.